tools/deploy.sh

```diff
@@ -90,39 +90,54 @@ RESTART_CONTAINER_CERTIFICATE_UPDATED=${RESTART_CONTAINER_CERTIFICATE_UPDATED:-"
 export NAMEPREFIX=${NAMEPREFIX:-"baremetal-operator"}
 
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+IRONIC_BASIC_AUTH_COMPONENT="${SCRIPTDIR}/ironic-deployment/components/basic-auth"
+TEMP_IRONIC_OVERLAY="${SCRIPTDIR}/ironic-deployment/overlays/temp"
+rm -rf "${TEMP_IRONIC_OVERLAY}"
+mkdir -p "${TEMP_IRONIC_OVERLAY}"
 
-KUSTOMIZE="tools/bin/kustomize"
+KUSTOMIZE="${SCRIPTDIR}/tools/bin/kustomize"
 make -C "$(dirname "$0")/.." "${KUSTOMIZE}"
 
+# Create a temporary overlay where we can make changes.
+pushd "${TEMP_IRONIC_OVERLAY}"
+${KUSTOMIZE} create --resources=../../../config/namespace \
+  --namespace=baremetal-operator-system --nameprefix=baremetal-operator-
+${KUSTOMIZE} edit add secret mariadb-password --from-literal=password=changeme
+
 if [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
     BMO_SCENARIO="${SCRIPTDIR}/config/basic-auth"
-    IRONIC_SCENARIO="${SCRIPTDIR}/ironic-deployment/basic-auth"
+    if [ "${DEPLOY_TLS}" == "true" ]; then
+        BMO_SCENARIO="${BMO_SCENARIO}/tls"
+        # Basic-auth + TLS is special since TLS also means reverse proxy, which affects basic-auth.
+        # Therefore we have an overlay that we use as base for this case.
+        ${KUSTOMIZE} edit add resource ../../overlays/basic-auth_tls
+    else
+        BMO_SCENARIO="${BMO_SCENARIO}/default"
+        ${KUSTOMIZE} edit add resource ../../base
+        ${KUSTOMIZE} edit add component ../../components/basic-auth
+    fi
 else
     BMO_SCENARIO="${SCRIPTDIR}/config"
-    IRONIC_SCENARIO="${SCRIPTDIR}/ironic-deployment"
-fi
-
-if [ "${DEPLOY_TLS}" == "true" ]; then
-    BMO_SCENARIO="${BMO_SCENARIO}/tls"
-    IRONIC_SCENARIO="${IRONIC_SCENARIO}/tls"
-elif [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
-    BMO_SCENARIO="${BMO_SCENARIO}/default"
+    if [ "${DEPLOY_TLS}" == "true" ]; then
```

Comment:
The specific combination of `[` and `==` should be avoided as `==` in non-Bash shell is undefined. So called bashism. For POSIX compliant scripting, use single `=` with single `[`. If you need to use Bash-only features, then make it `[[`. Bashisms may trigger unwanted errors in systems where default shell isn't Bash (for example Ubuntu often has dash).

Since this patch is not adding the line, feel free to not fix it in this patch, but avoid bashisms in future.